plugins/store/src/store.rs

```diff
@@ -428,19 +538,37 @@ impl<R: Runtime> Store<R> {
             .collect()
     }
 
+    /// Returns the number of elements in the store.
     pub fn length(&self) -> usize {
         self.store.lock().unwrap().len()
     }
 
+    /// Update the store from the on-disk state
     pub fn load(&self) -> crate::Result<()> {
-        self.store.lock().unwrap().load(self.deserialize)
+        self.store.lock().unwrap().load()
     }
 
+    /// Saves the store to disk at the store's `path`.
     pub fn save(&self) -> crate::Result<()> {
         if let Some(sender) = self.auto_save_debounce_sender.lock().unwrap().take() {
             let _ = sender.send(AutoSaveMessage::Cancel);
         }
-        self.store.lock().unwrap().save(self.serialize)
+        self.store.lock().unwrap().save()
+    }
+
+    /// Removes the store from the resource table,
+    /// this doesn't remove other references of the same store held by you,
+    /// and the store is only truely closed once all reference held by you are also dropped
+    pub fn close_store(self) {
+        let store = self.store.lock().unwrap();
+        let app = store.app.clone();
+        let state = app.state::<StoreState>();
+        let stores = state.stores.lock().unwrap();
+        if let Some(rid) = stores.get(&store.path).copied() {
+            drop(store);
+            drop(stores);
+            let _ = app.resources_table().take::<Store<R>>(rid);
+        }
```

Comment:
this should be a drop implementation instead

Reply:
This is to remove it from the resource table, we can't really do it in drop

Reply:
why not? this method already takes ownership of `self` so it can no longer be used which is what the drop implementation does.

Reply:
This is should be used from an `Arc` and since another reference is in the resource table, this wouldn't necessarily drop the `Store` I don't think

Reply:
taking `self` will not work with `Arc`, it will fail with compile error. You need to remove/unwrap the `Arc` in order to call this method. At which point, you can just drop it

Reply:
Got it, that makes sense

What's your suggestion on this, a static method that takes `Arc<Self>` instead or put the method on `Manager`? Or something else?

Reply:
if the idea is to only remove resources, just take `&self` and call it `close_resource` or something